.github/workflows/go.yml

```diff
@@ -17,4 +17,7 @@ jobs:
         with:
           go-version: 1.21.4
       - name: Run tests
-        run: go test -short $(go list ./... | grep -Pv 'go-ethereum/(accounts/keystore|eth/downloader|miner)$')
+        run: | # Upstream flakes are race conditions exacerbated by concurrent tests
+          FLAKY_REGEX='go-ethereum/(eth|accounts/keystore|eth/downloader|miner)$';
+          go list ./... | grep -P "${FLAKY_REGEX}" | xargs -n 1 go test -short;
```

Comment:
just a heads up that -P isn't supported on mac

Reply:
Thanks. I forgot because I very quickly installed GNU tools when I got one!

When we extend CI testing beyond this basic one we should discuss the extent to which we want to run upstream tests in different environments, given that they already do it extensively.